### PR TITLE
docs: add atomWithBroadcast and remove subheadings

### PR DESCRIPTION
**Pull request by gerzonc**
## Related Issues or Discussions

https://github.com/pmndrs/jotai/pull/2083#discussion_r1312498153

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/recipes/atom-with-default-refresh.mdx [hunk 1]
```diff
@@ -1,12 +1,10 @@
 ---
 title: atomWithRefreshAndDefault
```
Comment (dai-shi) [inline docs/recipes/atom-with-default-refresh.mdx:2]: Is it intentional to change the file name from the title?
Can it be `atom-with-refresh-and-default.mdx`?

Comment (dai-shi) [reply docs/recipes/atom-with-default-refresh.mdx:2]: Same for `atom-with-toggle-and-storage.mdx`

**Review by dai-shi — COMMENTED** [review]
I think it's good for consistency even though it's a bit long.

File: docs/recipes/atom-with-default-refresh.mdx (hunk 1 above)
Comment (gerzonc) [reply docs/recipes/atom-with-default-refresh.mdx:2]: That's a workaround for this issue I'm having when I name them like that

image

Comment (dai-shi) [reply docs/recipes/atom-with-default-refresh.mdx:2]: Not sure what you mean. My concern is the mdx file name.

Comment (gerzonc) [reply docs/recipes/atom-with-default-refresh.mdx:2]: The issue I mentioned before is that on the sidebar it toggles two items selected when you go to atomWithToggleAndStorage when you name the file like `atom-with-toggle-and-storage.mdx` (in the screenshot you can see that atomWithToggle and atomWithToggleAndStorage are selected). Same happens with `atom-with-refresh-and-default.mdx`. 

I'll update it right away, but just so you know about it :)

Comment (dai-shi) [reply docs/recipes/atom-with-default-refresh.mdx]: I see what you mean. It's the website issue. Let's merge this and fix it separately. @sandren can look into it?

**Review by dai-shi — APPROVED** [review]

**Pull request merged**